packages/rpc-methods/src/restricted/getBip32Entropy.ts

```diff
@@ -0,0 +1,260 @@
+  // `utils`, so this is just a workaround for testing.
+  const isPlainObject = (v: unknown): v is Record<string, unknown> => {
+    return true;
+  };
+
+  if (!isPlainObject(value)) {
+    throw ethErrors.rpc.invalidParams({
+      message: 'Expected a plain object.',
+    });
+  }
+
+  if (
+    !('path' in value) ||
+    !Array.isArray(value.path) ||
+    value.path.length === 0
+  ) {
+    throw ethErrors.rpc.invalidParams({
+      message: `Invalid "path" parameter. The path must be a non-empty BIP-32 derivation path array.`,
+    });
+  }
+
+  if (value.path[0] !== 'm') {
+    throw ethErrors.rpc.invalidParams({
+      message: `Invalid "path" parameter. The path must start with "m".`,
+    });
+  }
+
+  if (
+    value.path
+      .slice(1)
+      .some((v) => typeof v !== 'string' || !INDEX_REGEX.test(v))
+  ) {
+    throw ethErrors.rpc.invalidParams({
+      message: `Invalid "path" parameter. The path must be a valid BIP-32 derivation path array.`,
+    });
+  }
+
+  if (value.path.length < 3) {
+    throw ethErrors.rpc.invalidParams({
+      message: `Invalid "path" parameter. Paths must have at least two indices.`,
+    });
+  }
+
+  if (
+    !('curve' in value) ||
+    (value.curve !== 'secp256k1' && value.curve !== 'ed25519')
+  ) {
+    throw ethErrors.rpc.invalidParams({
+      message: `Invalid "curve" parameter. The curve must be "secp256k1" or "ed25519".`,
+    });
+  }
+
+  if (value.curve === 'ed25519' && value.path.some((v) => !v.endsWith("'"))) {
+    throw ethErrors.rpc.invalidParams({
+      message: `Invalid "path" parameter. Ed25519 does not support unhardened paths.`,
+    });
+  }
+}
+
+/**
+ * Validate the path values associated with a caveat. This validates that the
+ * value is a non-empty array with valid derivation paths and curves.
+ *
+ * @param caveat - The caveat to validate.
+ * @throws If the value is invalid.
+ */
+function validateCaveatPaths(caveat: Caveat<string, any>) {
+  if (
+    !('value' in caveat) ||
+    !Array.isArray(caveat.value) ||
+    caveat.value.length === 0
+  ) {
+    throw ethErrors.rpc.invalidParams({
+      message: 'Expected non-empty array of paths.',
+    });
+  }
+
+  caveat.value.forEach((path) => validatePath(path));
+}
+
+/**
+ * The specification builder for the `snap_getBip32Entropy` permission.
+ * `snap_getBip32Entropy` lets the Snap control private keys for a particular
+ * BIP-32 node.
+ *
+ * @param options - The specification builder options.
+ * @param options.allowedCaveats - The optional allowed caveats for the permission.
+ * @param options.methodHooks - The RPC method hooks needed by the method implementation.
+ * @returns The specification for the `snap_getBip32Entropy` permission.
+ */
+const specificationBuilder: PermissionSpecificationBuilder<
+  PermissionType.RestrictedMethod,
+  GetBip32EntropySpecificationBuilderOptions,
+  GetBip32EntropySpecification
+> = ({
+  allowedCaveats = null,
```

Comment:
Since we're being strict about the caveats of this permission in the permission validator I think we should remove this parameter.

Reply:
I wasn't sure what it was used for, so I kept it in for now, but you're right and I'll remove it.